src/code42cli/cmds/devices.py

```diff
@@ -0,0 +1,399 @@
+                backupDestination["lastCompletedBackup"]
+                for backupDestination in device["backupUsage"]
+            ]
+        )
+        if len(device["backupUsage"]) > 0
+        else None
+    )
+    return device
+
+
+@devices.group(cls=OrderedGroup)
+@sdk_options(hidden=True)
+def bulk(state):
+    """Tools for managing devices in bulk"""
+    pass
+
+
+@bulk.command(name="list", help="Get information about many devices")
+@click.option(
+    "--active",
+    required=False,
+    type=bool,
+    is_flag=True,
+    default=None,
+    help="Include to return only active devices.",
+)
+@click.option(
+    "--days-since-last-connected",
+    required=False,
+    type=int,
+    help="Return only devices that have not connected in the number of days specified.",
+)
+@click.option(
+    "--org-uid",
+    required=False,
+    type=str,
+    default=None,
+    help="Optionally provide to limit devices to only the ones in the org you specify. Note that child orgs will be included.",
+)
+@click.option(
+    "--drop-most-recent",
+    required=False,
+    type=int,
+    help="Will drop the X most recently connected devices for each user from the result list where X is the number you provide as this argument. Can be used to avoid passing the most recently connected device for a user to the deactivate command",
+)
+@click.option(
+    "--include-backup-usage",
+    required=False,
+    type=bool,
+    default=False,
+    is_flag=True,
+    help="Include to return backup usage information for each device (may significantly lengthen the size of the return)",
+)
+@click.option(
+    "--include-usernames",
+    required=False,
+    type=bool,
+    default=False,
+    is_flag=True,
+    help="Include to add the username associated with a device to the output",
+)
+@click.option(
+    "--include-settings",
+    required=False,
+    type=bool,
+    default=False,
+    is_flag=True,
+    help="Include to include device settings in output",
+)
+@sdk_options()
+def bulk_list(
+    state,
+    active,
+    days_since_last_connected,
+    drop_most_recent,
+    org_uid,
+    include_backup_usage,
+    include_usernames,
+    include_settings,
+):
+    """Outputs a list of all devices in the tenant"""
+    devices_dataframe = _get_device_dataframe(
+        state.sdk, active, org_uid, include_backup_usage
+    )
+    if drop_most_recent:
+        devices_dataframe = _drop_n_devices_per_user(
+            devices_dataframe, drop_most_recent
+        )
+    if days_since_last_connected:
+        devices_dataframe = _drop_devices_which_have_not_connected_in_some_number_of_days(
+            devices_dataframe, days_since_last_connected
+        )
+    if include_settings:
+        devices_dataframe = _add_settings_to_dataframe(state.sdk, devices_dataframe)
+    if include_usernames:
+        devices_dataframe = _add_usernames_to_device_dataframe(
+            state.sdk, devices_dataframe
+        )
+    click.echo(devices_dataframe.to_csv())
```

Comment:
When output is large (greater than 10), we have been using `click.echo_via_pager()`. Would that make sense here?

Reply:
Yes, the output could be quite long. Updated.